### ISSUE TITLE
Add Metadata to Service Registration in Consul

### ISSUE THREAD

**Issue by ballisticpotato** [hashicorp/vault#9121]
**Is your feature request related to a problem? Please describe.**
We have some internal automation that sets up networking for our Consul services based on the metadata attached to our services. Vault currently doesn't have the option to set metadata in the `service_registration` stanza and so we can't use our internal automation to setup additional networking.

**Describe the solution you'd like**
We would like the ability to configure the metadata for when Vault registers itself with Consul. This could by done by extending the `service_registration` stanza for using Consul.

```
service_registration "consul" {
    metadata = {
        tag1 = "value1"
        tag2 = "value2"
    }
 }
```

**Describe alternatives you've considered**
We've tried updating Vault self registration using the Consul HTTP API, but realized that Vault would override those updates.

We could also disable to self registration, but we would then lose some of the automatic configuration of the HA redirect address.

We've resorted to manually setting up our three Vault services in our networking and it works for now, but we'd like to bring Vault into the folds of our existing automation.

**Explain any additional use-cases**
This is a nice to have for customizing Vault's self registration. There is already the option for setting `service_tags` in the `consul` service_registration parameters.

**Additional context**
None

**Comment by voiprodrigo** [comment]
I second this, it would be beneficial for my automation also, in parallel to service_tags.

**Comment by rrijkse** [comment]
Any update on this? There is an open PR that seems like it would work. This would be very useful for us to have as well.
